### PR TITLE
Add async-profiler to Hadoop and HBase

### LINKED ISSUE #532: Include async-profiler v2 into the HDFS & HBase images
This allows users to start a _profile_ run for those tools from the Web UI which makes it much easier for support staff and administrators to diagnose performance issues.

Part of this work will be [upstreamed](https://issues.apache.org/jira/browse/HBASE-28242) at HBase but only for the upcoming 3.x version.
We still need to propose the same changes to the HDFS project.

The current work can be found in the `SUP-56` branch.

Tasks left to do (might be incomplete):
- [x] Add a test which checks whether the profiler servlet is available (only 200/404 status check)
- [x] Remove hardcoded version numbers
- [x] Mirror the async-profiler binary to our own repository
- [x] Remove workarounds to make the images compatible with 23.7

### PR DESCRIPTION
**Pull request by siegfriedweber**
# Description

Add async-profiler to Hadoop and HBase

Closes #532 

## Integration tests

* stackabletech/hbase-operator#445
  https://ci.stackable.tech/view/02%20Operator%20Tests%20(custom)/job/hbase-operator-it-custom/137/
* stackabletech/hdfs-operator#465
  https://ci.stackable.tech/view/02%20Operator%20Tests%20(custom)/job/hdfs-operator-it-custom/117/

## Definition of Done Checklist

- Not all of these items are applicable to all PRs, the author should update this template to only leave the boxes in that are relevant
- Please make sure all these things are done and tick the boxes
 
```[tasklist]
- [x] Changes are OpenShift compatible
- [x] All added packages (via microdnf or otherwise) have a comment on why they are added
- [x] Things not downloaded from Red Hat repositories should be mirrored in the Stackable repository and downloaded from there
```

### CODE REVIEW

File: hadoop/Dockerfile [hunk 1]
```diff
@@ -2,9 +2,11 @@
 FROM stackable/image/java-base AS builder
 
 ARG PRODUCT
+ARG ASYNC_PROFILER
 ARG JMX_EXPORTER
 ARG PROTOBUF
 ARG TOPOLOGY_PROVIDER
+ARG TARGETARCH
```
[the remark this answers is not in the document]
Comment (siegfriedweber) [reply hadoop/Dockerfile:9]: The environment variable `TARGETARCH` is set by `docker buildx` either implicitly or explicitly with the parameter `--platform`, see https://docs.docker.com/build/guide/multi-platform/#platform-build-arguments. I tested this approach successfully with `amd64`. I hope that it also works with `arm64`.

[the remark this answers is not in the document]
Comment (lfrancke) [reply hadoop/Dockerfile:9]: That's good to know.
This is how Maxi does it: https://github.com/stackabletech/docker-images/pull/534/files#diff-938a5aed69f272f4d0a7f6ab596ce9bbf1b2dda19ce600a7c190384924431f2eR61

I don't have a preference but I would prefer if you could coordinate to chose one approach.

File: hadoop/Dockerfile [hunk 2]
```diff
@@ -36,6 +38,16 @@ RUN curl --fail "https://repo.stackable.tech/repository/packages/jmx-exporter/jm
...
     ln -s /stackable/jmx/jmx_prometheus_javaagent.jar /stackable/jmx/jmx_prometheus_javaagent-0.16.1.jar
 
+RUN if [ "$TARGETARCH" = "amd64" ]; then \
+      arch=x64; \
+    elif [ "$TARGETARCH" = "arm64" ]; then \
+      arch=arm64; \
+    else \
+      exit 1; \
+    fi; \
+    curl --fail -L https://repo.stackable.tech/repository/packages/async-profiler/async-profiler-${ASYNC_PROFILER}-linux-$arch.tar.gz  | tar -xzC . && \
```
[the remark this answers is not in the document]
Comment (lfrancke) [reply hadoop/Dockerfile:48]: Thank you for checking.

[the remark this answers is not in the document]
Comment (dervoeti) [reply hadoop/Dockerfile:48]: Yeah I think this is nice, we talked about this today together with @Maleware, I'd say including the .jar itself is sufficient for now. We could include the SBOMs of the binaries we pull in as well and I have some ideas about referring to external SBOMs using [bomlink](https://cyclonedx.org/capabilities/bomlink/), but that would be a next step. I think currently there's no need to include the SBOMs of stuff we pull in, just make sure that the stuff we pull in itself is present in the final SBOM.

File: hadoop/Dockerfile (hunk 1 above)
[the remark this answers is not in the document]
Comment (siegfriedweber) [reply hadoop/Dockerfile:9]: We decided to use `$TARGETARCH` with shell parameter expansion, e.g.

```shell
ARCH="${TARGETARCH/amd64/x86_64}" ARCH="${ARCH/arm64/aarch64}"
```

I adapted this pull request in 2a69d380c4196518a9560ddbd84905e6677ab574:

```Dockerfile
RUN ARCH="${TARGETARCH/amd64/x64}" && \
    curl --fail -L "https://repo.stackable.tech/repository/packages/async-profiler/async-profiler-${ASYNC_PROFILER}-${TARGETOS}-${ARCH}.tar.gz"  | tar -xzC . && \
    ln -s "/stackable/async-profiler-${ASYNC_PROFILER}-${TARGETOS}-${ARCH}" /stackable/async-profiler
```

@Maleware will adapt the other occurrences in another pull request.

**Review by lfrancke — APPROVED** [review]

**Pull request merged**